### PR TITLE
Add toArray() method to map lang library

### PR DESCRIPTION
**Pull request by pubudu91**
## Samples
```ballerina
// Getting the values as an array
map<int> ints = {"one": 1, "two": 2, "three": 3, "four": 4, "five": 5};
int[] intArr = ints.toArray();

// Getting record field values as an array
type Foo record {|
    string name;
    int age;
    float weight;
    decimal height;
    boolean isStudent;
|};

Foo foo = {
        name: "John Doe",
        age: 25,
        weight: 65.5,
        height: 172.3,
        isStudent: true
    };

(string|int|float|decimal|boolean)[] arr = foo.toArray();
```
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/lang.map/src/main/java/org/ballerinalang/langlib/map/ToArray.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+@BallerinaFunction(
+        orgName = "ballerina", packageName = "lang.map",
+        functionName = "toArray",
+        args = {@Argument(name = "m", type = TypeKind.MAP)},
+        returnType = {@ReturnType(type = TypeKind.ARRAY, elementType = TypeKind.ANY)},
+        isPublic = true
+)
+public class ToArray {
+
+    public static ArrayValue toArray(Strand strand, MapValue<?, ?> m) {
```
Comment (pramodya1994) [inline langlib/lang.map/src/main/java/org/ballerinalang/langlib/map/ToArray.java]: Can we add a test for this method?

Comment (pubudu91) [reply langlib/lang.map/src/main/java/org/ballerinalang/langlib/map/ToArray.java]: Added with 11bc310

File: langlib/langlib-test/src/test/resources/test-src/maplib_test.bal [hunk 2]
```diff
@@ -73,3 +73,140 @@ function testReduce() returns float {
...
+function testMapOfUnionToArray() {
+    map<int|string|Person> m = {"i": 10, "s": "foo", "p": new Person("Pubudu")};
+    (int|string|Person)[] arr = m.toArray();
+
+    assert(10, <anydata>arr[0]);
+    assert("foo", <anydata>arr[1]);
+    assertSameRef(m["p"], arr[2]);
+}
+
+type Foo record {|
```
Comment (MaryamZi) [inline langlib/langlib-test/src/test/resources/test-src/maplib_test.bal]: Shall we also add a test for records where the record fields are all of a single simple basic type?

```ballerina
type Bar record {|
    byte a;
    byte b;
    byte...;
|};
```

This array can also be updated right? Shall we add a test or two to cover updating a value at an index and maybe adding new elements at the end?

File: langlib/langlib-test/src/test/resources/test-src/maplib_test.bal [hunk 3]
```diff
@@ -73,3 +73,140 @@ function testReduce() returns float {
...
+    var arr = foo.toArray();
+
+    assert(<(string|int|float|decimal|boolean)[]>["John Doe", 25, 65.5, 172.3d, true, "Sri Lanka", 12500], arr);
+}
+
+
+// Util functions
+
+function assert(anydata expected, anydata actual) {
+    if (expected != actual) {
```
Comment (MaryamZi) [inline langlib/langlib-test/src/test/resources/test-src/maplib_test.bal]: Shall we return here after checking `==` instead?

File: langlib/langlib-test/src/test/resources/test-src/maplib_test.bal [hunk 4]
```diff
@@ -73,3 +73,140 @@ function testReduce() returns float {
...
+        typedesc<anydata> expT = typeof expected;
+        typedesc<anydata> actT = typeof actual;
+        string reason = "expected [" + expected.toString() + "] of type [" + expT.toString()
+                            + "], but found [" + actual.toString() + "] of type [" + actT.toString() + "]";
+        error e = error(reason);
+        panic e;
```
Comment (MaryamZi) [inline langlib/langlib-test/src/test/resources/test-src/maplib_test.bal]:
```suggestion
        panic error(reason);
```

Just to be consistent with our usage of reasons, shall we use a simple string as the reason (e.g., "AssertionError") and then add this current reason as the message field of the detail?

**Pull request merged**